File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        int wr = 0;
        int rd = 1;

        while(rd != n){
            if(nums[wr] != nums[rd]){
                wr++;
                nums[wr] = nums[rd];
                rd++;
            }
            else
                rd++;
        }

        return wr+1;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.swift
class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        let n = nums.count

        if n == 0 {return 0}
        var wd = 0

        for rd in 1..<n {
            if nums[rd] != nums[wd] {
                wd += 1
                nums[wd] = nums[rd]
            }
        }

        return wd+1
    }
}